colossalai/utils/memory_tracer/memstats_collector.py

```diff
@@ -107,9 +82,37 @@ def start_collection(self):
         self._mem_monitor.start()
 
     def finish_collection(self):
+        self.sample_overall_data()
+        self._step_total = len(self._sampling_time)
         self._start_flag = False
         self._mem_monitor.finish()
 
+    def sample_model_data(self) -> None:
+        """Sampling model data statistics.
+        """
+        if self._start_flag:
+            cuda_mem, cpu_mem = GLOBAL_MODEL_DATA_TRACER.both_mem_usage
+            self._model_data_cuda_list.append(cuda_mem)
+            self._model_data_cpu_list.append(cpu_mem)
+
+    def sample_overall_data(self) -> None:
+        """Sampling non model data statistics.
+        """
+        if self._start_flag:
+            # overall data recording is after model data recording
+            if len(self._model_data_cuda_list) == 0:
+                return
+
+            self._overall_cuda_list.append(self._mem_monitor.finish())
+            self._overall_cpu_list.append(colo_device_memory_used(torch.device('cpu')))
+
+            assert len(self._model_data_cuda_list) == len(self._overall_cuda_list)
+
+            self._non_model_data_cuda_list.append(self._overall_cuda_list[-1] - self._model_data_cuda_list[-1])
+            self._non_model_data_cpu_list.append(self._overall_cpu_list[-1] - self._model_data_cpu_list[-1])
+            self._sampling_time.append(time.time())
+            self._mem_monitor.start()
+
     def sample_memstats(self) -> None:
```

Comment:
Where do you call this method?

Reply:
Currently, test_stateful_tensor_mgr.py.
I will change the code in test_stateful_tensor_mgr.py soon.
So it will appear in nowhere.
But I just keep it for some recording operations.